x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -0,0 +1,753 @@
+        String lastIncrementalSyncScheduledAt,
+        Long lastIndexedDocumentCount,
+        String lastSeen,
+        String lastSyncError,
+        String lastSyncScheduledAt,
+        ConnectorSyncStatus lastSyncStatus,
+        String lastSynced,
+        String name,
+        ConnectorIngestPipeline pipeline,
+        ConnectorScheduling scheduling,
+        String serviceType,
+        ConnectorStatus status,
+        Object syncCursor,
+        boolean syncNow
+    ) {
+        this.connectorId = connectorId;
+        this.apiKeyId = apiKeyId;
+        this.configuration = configuration;
+        this.customScheduling = customScheduling;
+        this.description = description;
+        this.error = error;
+        this.features = features;
+        this.indexName = indexName;
+        this.isNative = isNative;
+        this.language = language;
+        this.lastAccessControlSyncError = lastAccessControlSyncError;
+        this.lastAccessControlSyncScheduledAt = lastAccessControlSyncScheduledAt;
+        this.lastAccessControlSyncStatus = lastAccessControlSyncStatus;
+        this.lastDeletedDocumentCount = lastDeletedDocumentCount;
+        this.lastIncrementalSyncScheduledAt = lastIncrementalSyncScheduledAt;
+        this.lastIndexedDocumentCount = lastIndexedDocumentCount;
+        this.lastSeen = lastSeen;
+        this.lastSyncError = lastSyncError;
+        this.lastSyncScheduledAt = lastSyncScheduledAt;
+        this.lastSyncStatus = lastSyncStatus;
+        this.lastSynced = lastSynced;
+        this.name = name;
+        this.pipeline = pipeline;
+        this.scheduling = scheduling;
+        this.serviceType = serviceType;
+        this.status = status;
+        this.syncCursor = syncCursor;
+        this.syncNow = syncNow;
+    }
+
+    public Connector(StreamInput in) throws IOException {
+        this.connectorId = in.readString();
+        this.apiKeyId = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.configuration = in.readMap(StreamInput::readString, StreamInput::readGenericValue);
+        } else {
+            this.configuration = null;
+        }
+        if (in.readBoolean()) {
+            this.customScheduling = new ConnectorCustomSchedule(in);
+        } else {
+            this.customScheduling = null;
+        }
+        this.description = in.readOptionalString();
+        this.error = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.features = new ConnectorFeatures(in);
+        } else {
+            this.features = null;
+        }
+        this.indexName = in.readOptionalString();
+        this.isNative = in.readBoolean();
+        this.language = in.readOptionalString();
+        this.lastAccessControlSyncError = in.readOptionalString();
+        this.lastAccessControlSyncScheduledAt = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.lastAccessControlSyncStatus = ConnectorSyncStatus.valueOf(in.readString());
```

Comment:
readOptionalWriteable would work here as well. Applicable to other classes down below 👍